modules/openapi-generator/src/main/resources/typescript/types/PromiseAPI.mustache

```diff
@@ -55,21 +55,7 @@ export class Promise{{classname}} {
      {{/allParams}}
      */
     public {{nickname}}WithHttpInfo({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}_options?: {{#useInversify}}Configuration{{/useInversify}}{{^useInversify}}PromiseConfigurationOptions{{/useInversify}}): Promise<HttpInfo<{{{returnType}}}{{^returnType}}void{{/returnType}}>> {
-        let observableOptions: undefined | Configuration{{^useInversify}}Options{{/useInversify}}{{#useInversify}} = _options{{/useInversify}}
-        {{^useInversify}}
-        if (_options){
-	    observableOptions = {
-                baseServer: _options.baseServer,
-                httpApi: _options.httpApi,
-                middleware: _options.middleware?.map(
-                    m => new PromiseMiddlewareWrapper(m)
-		),
-		middlewareMergeStrategy: _options.middlewareMergeStrategy,
-                authMethods: _options.authMethods
-	    }
-	}
-        {{/useInversify}}
-        const result = this.api.{{nickname}}WithHttpInfo({{#allParams}}{{paramName}}, {{/allParams}}observableOptions);
+        const result = this.api.{{nickname}}WithHttpInfo({{#allParams}}{{paramName}}, {{/allParams}}{{#useInversify}}_options{{/useInversify}}{{^useInversify}}wrapOptions(_options){{/useInversify}});
```

Comment:
After looking at the diff below, should we keep the `observableOptions` local, that we then pass to the api call?

i.e. instead of
```ts
const result = this.api.listWithHttpInfo(wrapOptions(_options));
```

we would have:
```ts
const observableOptions = wrapOptions(_options)
const result = this.api.listWithHttpInfo(observableOptions);
```
this would reduce the diff a tiny bit, keep the api call separate from the wrap call and still convey the logic of what happens in the wrappng.

Take or leave.

Reply:
Not quite sure what the goal is.
The samples are ok like this I think.
If you want to make the template more readable, we could wrap the parameters on separate lines.

Reply:
I am not hooked on this, but basically two goals:

* reduce the diff in this pull request, as the second line would stay as-is if we don't inline and/or change the variable name
* reduce template complexity - the less we nest and wrap, etc. the easier to maintain. Extra lines in the output will be minified away. The current change brings the length of this line from 120 characters to 200 characters, which means it's almost twice as long, possible more than twice as complex with all the `useInversify` conditionals.